src/migrateStaticObservableMethodsRule.ts

```diff
@@ -0,0 +1,182 @@
+// Original author Bowen Ni
+// Modifications mgechev.
+
+import * as Lint from 'tslint';
+import * as tsutils from 'tsutils';
+import * as ts from 'typescript';
+import { subtractSets, concatSets, isObservable, returnsObservable, computeInsertionIndexForImports } from './utils';
+/**
+ * A typed TSLint rule that inspects observable chains using patched RxJs
+ * operators and turns them into a pipeable operator chain.
+ */
+export class Rule extends Lint.Rules.TypedRule {
+  static metadata: Lint.IRuleMetadata = {
+    ruleName: 'migrate-static-observable-methods',
+    description: 'Updates the static methods of the Observable class.',
+    optionsDescription: '',
+    options: null,
+    typescriptOnly: true,
+    type: 'functionality'
+  };
+  static IMPORT_FAILURE_STRING = 'prefer operator imports with no side-effects';
+  static OBSERVABLE_FAILURE_STRING = 'prefer function calls';
+
+  applyWithProgram(sourceFile: ts.SourceFile, program: ts.Program): Lint.RuleFailure[] {
+    const failure = this.applyWithFunction(sourceFile, ctx => this.walk(ctx, program));
+    return failure;
+  }
+  private walk(ctx: Lint.WalkContext<void>, program: ts.Program) {
+    this.removePatchedOperatorImports(ctx);
+    const sourceFile = ctx.sourceFile;
+    const typeChecker = program.getTypeChecker();
+    const insertionStart = computeInsertionIndexForImports(sourceFile);
+    let rxjsOperatorImports = findImportedRxjsOperators(sourceFile);
+
+    function checkPatchableOperatorUsage(node: ts.Node) {
+      if (!isRxjsStaticOperatorCallExpression(node, typeChecker)) {
+        return ts.forEachChild(node, checkPatchableOperatorUsage);
+      }
+
+      const callExpr = node as ts.CallExpression;
+      if (!tsutils.isPropertyAccessExpression(callExpr.expression)) {
+        return ts.forEachChild(node, checkPatchableOperatorUsage);
+      }
+
+      const propAccess = callExpr.expression as ts.PropertyAccessExpression;
+      const name = propAccess.name.getText(sourceFile);
+      const operatorName = OPERATOR_RENAMES[name] || name;
+      const start = propAccess.getStart(sourceFile);
+      const end = propAccess.getEnd();
+      const operatorsToImport = new Set<string>([operatorName]);
```

Comment:
nit: Consider using `operatorAlias(operatorName)` here to handle the edge case when the operator is already imported as an alias (half migrated code).